source/drivers/payload.cpp

```diff
@@ -26,7 +27,40 @@ err_t handlePayload(uint8_t* buffer, uint8_t buffer_len) {
         image_checksum = buffer[9] << 24 | buffer[10] << 16 | buffer[11] << 8 |
                          buffer[12] << 0;
 
-        // TODO Validate image base address?
+        // Confirm the CRC is valid
+        // Reset the CRC module to clear out any previous calcuations
+        init_crc();
+
+        uint32_t program_counter = 0;
+
+        while (program_counter < flight_software_length) {
+            uint8_t image_bytes[32];
+
+            uint8_t bytes_read = 0;
+            err_t bytes_error =
+                getProgramBytes((ImageBaseAddress)image_base_address,
```

Comment:
**Reporter**: CPPCHECK
**Rule**: unreadVariable
**Severity**: INFO
**File**: source/drivers/payload.cpp L41

Variable &#39;bytes_error&#39; is assigned a value that is never used.. Variable &#39;bytes_error&#39; is assigned a value that is never used.


*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a2085128301>*